--- origsrc/mate-document-viewer-1.2.1/help/reference/shell/atril.types	2012-03-15 05:37:04.000000000 -0500
+++ src/mate-document-viewer-1.2.1/help/reference/shell/atril.types	2012-04-17 14:07:09.212504800 -0500
@@ -2,7 +2,6 @@ egg_find_bar_get_type
 ev_application_get_type
 ev_file_monitor_get_type
 ev_history_get_type
-ev_media_player_keys_get_type
 ev_message_area_get_type
 ev_metadata_get_type
 ev_navigation_action_get_type
