--- origsrc/atril-1.20.1/libview/ev-jobs.c	2018-03-22 10:13:28.000000000 -0500
+++ src/atril-1.20.1/libview/ev-jobs.c	2018-04-23 00:18:34.417415900 -0500
@@ -876,8 +876,6 @@ snapshot_callback(WebKitWebView *webview
 	gtk_widget_destroy (gtk_widget_get_toplevel (GTK_WIDGET (webview)));
 }
 
-#endif  /* ENABLE_EPUB */
-
 static void
 web_thumbnail_get_screenshot_cb (WebKitWebView  *webview,
                                  WebKitLoadEvent event,
@@ -910,6 +908,8 @@ webview_load_failed_cb (WebKitWebView  *
 	return TRUE;
 }
 
+#endif  /* ENABLE_EPUB */
+
 static gboolean
 ev_job_thumbnail_run (EvJob *job)
 {
